src/services/HandTrackingService/HandTrackingService.ts

```diff
@@ -0,0 +1,97 @@
+import type { Coordinates3D, HandVector, IHandTrackingService } from './HandTrackingService.type';
+import { Hands, HandsInterface, Options, Results, ResultsListener } from '@mediapipe/hands';
+import { CanNotFindCameraError } from '../../errors/CanNotFindCameraError';
+import { CanNotPerformPredictionError } from '../../errors/CanNotPerformPredictionError';
+import { isDefined } from '../../utils/isDefined';
+import { ServiceUnavailableError } from '../../errors/ServiceUnavailableError';
+
+export class HandTrackingService implements IHandTrackingService {
+  private static MODEL_SOURCE = 'https://cdn.jsdelivr.net/npm/@mediapipe/hands/';
+  private isActive: boolean;
+  private isInitialized: boolean;
+  private videoSource: MediaStream | undefined;
+  private handCoordinates: Coordinates3D[][] = [];
+
+  private readonly hands: HandsInterface;
+  private readonly videoElement: HTMLVideoElement;
+
+  constructor(videoSource?: MediaStream, settings?: Options) {
+    this.isActive = false;
+    this.isInitialized = false;
+    this.videoElement = document.createElement('video');
+    this.videoSource = videoSource;
+    this.hands = new Hands({
+      locateFile: (file) => {
+        return `${HandTrackingService.MODEL_SOURCE}${file}`;
+      },
+    });
+    this.hands.setOptions({
+      maxNumHands: 1,
+      minDetectionConfidence: 0.5,
+      minTrackingConfidence: 0.5,
+      modelComplexity: 1,
```

Comment:
Won't this be classified as "magic numbers"?

Reply:
nope, if this would be a 'magic number' our pr-check pipeline would fail